pkg/cmd/root/extension.go

```diff
@@ -4,20 +4,45 @@ import (
 	"errors"
 	"fmt"
 	"os/exec"
+	"strings"
 
 	"github.com/cli/cli/v2/pkg/extensions"
 	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/mgutz/ansi"
 	"github.com/spf13/cobra"
 )
 
 type ExternalCommandExitError struct {
 	*exec.ExitError
 }
 
+type extensionReleaseInfo struct {
+	CurrentVersion string
+	LatestVersion  string
+	Pinned         bool
+	URL            string
+}
+
 func NewCmdExtension(io *iostreams.IOStreams, em extensions.ExtensionManager, ext extensions.Extension) *cobra.Command {
+	// Setup channel containing information about potential latest release info
+	updateMessageChan := make(chan *extensionReleaseInfo)
+
 	return &cobra.Command{
 		Use:   ext.Name(),
 		Short: fmt.Sprintf("Extension %s", ext.Name()),
+		// PreRun handles looking up whether extension has a latest version only when the command is ran.
+		PreRun: func(c *cobra.Command, args []string) {
+			go func() {
+				if ext.UpdateAvailable() {
+					updateMessageChan <- &extensionReleaseInfo{
+						CurrentVersion: ext.CurrentVersion(),
+						LatestVersion:  ext.LatestVersion(),
+						Pinned:         ext.IsPinned(),
+						URL:            ext.URL(),
+					}
+				}
+			}()
```

Comment:
🤔 Something about this being an IIFE (instantly invoked functional expression) is giving me pause... That said, I'm not sure I've ever seen a goroutine created with a named function. Is that the convention?

Either way, I know you're interested in testing, and if this IIFE was extracted to a named function with its dependencies injected (`updateMessageChan`, `extensionReleaseInfo`, and `ext`) testing it would be pretty simple. It's a lot harder to test it here, as written.

Reply:
Just updated the PR with tests.  I don't think there's a lot of value testing channel handling behavior so much as if we're emitting the expected messages on standard error.

Reply:
Yeah, I agree... I didn't know the mocks existed for extensions already, so the tests for this are more straight forward than I anticipated ❤️ Definitely think you've nailed the tests